Source/WebCore/platform/graphics/SystemFontDatabase.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "FontSelectionAlgorithm.h"
+#include <array>
+#include <optional>
+#include <wtf/text/AtomString.h>
+
+namespace WebCore {
+
+class SystemFontDatabase {
+public:
+    WEBCORE_EXPORT static SystemFontDatabase& singleton();
+
+    enum class FontShorthand {
+        // This needs to be kept in sync with CSSValue.
+        Caption,
+        Icon,
+        Menu,
+        MessageBox,
+        SmallCaption,
+        WebkitMiniControl,
+        WebkitSmallControl,
+        WebkitControl,
+#if PLATFORM(COCOA)
+        AppleSystemHeadline,
+        AppleSystemBody,
+        AppleSystemSubheadline,
+        AppleSystemFootnote,
+        AppleSystemCaption1,
+        AppleSystemCaption2,
+        AppleSystemShortHeadline,
+        AppleSystemShortBody,
+        AppleSystemShortSubheadline,
+        AppleSystemShortFootnote,
+        AppleSystemShortCaption1,
+        AppleSystemTallBody,
+        AppleSystemTitle0,
+        AppleSystemTitle1,
+        AppleSystemTitle2,
+        AppleSystemTitle3,
+        AppleSystemTitle4,
+#endif
+        StatusBar,
+    };
+    using FontShorthandUnderlyingType = std::underlying_type<FontShorthand>::type;
+    static constexpr auto fontShorthandCount = static_cast<FontShorthandUnderlyingType>(FontShorthand::StatusBar) - static_cast<FontShorthandUnderlyingType>(FontShorthand::Caption) + 1;
```

Comment:
Not asking you to change this, and I guess it's not prevailing WebKit style to do this (only see a few instances of it), but I prefer the idiom of adding a `Count` value as the last value of an enum, when that's a thing that's needed. It would remove the risk of `fontShorthandCount` not being updated correctly. (I know that this means you'll get some non-exhaustive switch statement errors for that new value, but I think adding a dummy case for `Count` is a reasonable price to pay.)

I wouldn't bother subtracting the `Caption` value, if these enums won't have explicitly signed values.

Reply:
I actually think I disagree 😕 I guess kind of feel that A) the whole point of enums is to restrict the representable values of an `int` to just the ones that are valid, but if you add new values which aren't valid, you've lost the whole philosophical purpose of enums, and B) more pragmatically, I'm more bothered by extra cases in switches (or the siren's song of just using `default` instead) than I am about having some things in WebKit have to match other things - which is already true about lots of things in WebKit, including some other things in this patch itself.